### ISSUE TITLE
Make adapter.js more granular

### ISSUE THREAD

**Issue by juberti** [webrtcHacks/adapter#6]
From https://github.com/webrtc/samples/issues/134

Currently, adapter.js uses navigator.mozGetUserMedia to detect the browser. As adapter.js expands to support more environments, we should make the feature detection and injection more granular, so that, for example, RTCPeerConnection can be handled even in environments (e.g. Firefox Add-ons, node.js) that do not have navigator.getUserMedia.

**Comment by DamonOehlman** [comment]
Once again, happy to pitch in where it makes sense.  We've certainly tried to apply a very disciplined, modular approach over in rtc.io land.  Specifically here is a module where we do cross-browser detection for various browser-prefixed functions / classes:

https://github.com/rtc-io/rtc-core/blob/master/detect.js

Example Usage is something like:

``` js
var detect = require('rtc-core/detect');
var RTCPeerConnection = detect('RTCPeerConnection');
```

I'm sure a lot of folks from the JS community would be very happy to help out and lend some time to adapter.js being regeared for better reusability.  It's completely doable, and with tools like [browserify](https://github.com/substack/node-browserify) we could get to a point where certain pieces of adapter.js are "cherry-pickable" for those who want to go that way, and then [bundled](https://github.com/substack/browserify-handbook#standalone) for those who just want the whole box and dice.

I've read some of the history around this on #8 and understand that adapter.js really was just built to support a nice suite of WebRTC examples, so it won't receive any criticism from me (well maybe a little).  My problem is the amount of people just using what is put in front of them without considering how to properly structure an application...

**Comment by KaptenJansson** [comment]
What's our latest take on this based on recent learning's? @alvestrand , @jan-ivar, @fippo  WDYT?

**Comment by KaptenJansson** [comment]
PTAL #164 regarding this, any input is welcome.

**Comment by KaptenJansson** [comment]
Fixed in #164. Please file new specific issues or PR's if something should be changed.
